### ISSUE TITLE
MouseListener - MouseDragged

### ISSUE THREAD

**Issue by cupstudio** [craftworkgames/MonoGame.Extended#94]
My initial impression of the MouseDragged (without looking at the code) would be that I could hookup with that event to move things around on the screen via the mouse.

After playing with it a bit, it looks like it only triggers once the mouse button is released.  This is ok too, I just need to actually use the `MouseMoved` event and track which object should be moving myself.  (`MouseDown` - select object - `MouseMoved` - move object if object still selected - `MouseUp` - de-select object)

It does make me curious about the intent of the `MouseDragged` event, however.  If we only get the event triggered on a button release it doesn't seem much more useful than `MouseUp`.  Distance is another issue - when moving an object you would need to get updates on the mouse movement regularly in order to move with the mouse.  Currently `MouseDragged` will only have the position data for this loop and previous loop and will only trigger at the very end.

I may be missing your intent with this event, however.  Please feel free to correct me!

**Comment by craftworkgames** [comment]
I wasn't the original author of the `MouseListener` class so I can't really comment on the intent of the event.

Looking at the code, it looks like the only difference between the `MouseDragged` and `MouseUp` events is that the dragged event only occurs if the mouse moves while the button is being held down. In that sense it appears to be similar to how the [`DragDrop` event](https://msdn.microsoft.com/en-us/library/system.windows.forms.control.dragdrop(v=vs.71).aspx) works in windows forms.

> Occurs when a drag-and-drop operation is completed.

However, looking at other libraries like [Unity `OnMouseDrag`](http://docs.unity3d.com/ScriptReference/MonoBehaviour.OnMouseDrag.html) it makes sense that an event should be called every frame in a game,

> OnMouseDrag is called every frame while the mouse is down.

LibGDX has a similar concept in their [`DragListener`](https://libgdx.badlogicgames.com/nightlies/docs/api/com/badlogic/gdx/scenes/scene2d/utils/DragListener.html) which is an input listener purely for drag drop operations. They have `dragStart`, `dragStop` and `drag` events and interestingly also `touchDown`, `touchUp` and `touchDragged` which works on both the mouse and touch screens.

> Called when a mouse button or a finger touch is moved anywhere, but only if touchDown previously returned true for the mouse button or touch.

So I think MonoGame.Extended is indeed lacking in this area. I propose we introduce a few new events and perhaps get rid of some of the old ones. At this stage I don't think it will be a big deal to break the API so long as we make it obvious that something has changed. 

In other words, I think changing the behavior of the `MouseDragged` event could really confuse people already using it. On the other hand if we removed `MouseDragged` introduced some new events like `DragStarted`, `DragEnded` and `Dragging` then at least people would get an easy to fix compile error.

Or alternately we could implement a `DragListener` like LibGDX. What do you think?
